bot.py

```diff
@@ -7,43 +7,184 @@
+            unknown. The value will be an error message displayable to the user which
+            says the command was not recognized.
+        """
+        if message_content[0] != "!":
+            raise ValueError(
+                f"Message '{message_content}' does not begin with '{MusicBot.COMMAND_PREFIX}'"
+            )
+
+        content_split = message_content[1:].split(" ", 1)
+        command_name = content_split[0]
+        command_content = ""
+        if len(content_split) == 2:
+            command_content = content_split[1]
+
+        if command_name not in self.handlers:
+            return None, None, f"Command {command_name} not recognized."
+
+        return self.handlers[command_name], command_content, None
+
+    async def hello(self, message, command_content):
+        await message.channel.send("Hello!")
+
+    async def countdown(self, message, command_content):
+        try:
+            seconds = int(command_content)
+            while seconds > 0:
+                await message.channel.send(seconds)
+                await asyncio.sleep(1)
+                seconds -= 1
+            await message.channel.send("BOOOM!!!")
+        except ValueError:
+            await message.channel.send(f"{command_content} is not an integer.")
+
+    async def dinkster(self, message, command_content):
+        for channel in await message.guild.fetch_channels():
+            if isinstance(channel, discord.VoiceChannel):
+                voice_client = await channel.connect()
+                audio_source = await discord.FFmpegOpusAudio.from_probe("Dinkster.ogg")
+                voice_client.play(audio_source)
+                await asyncio.sleep(10)
+                await voice_client.disconnect()
+
+    async def play(self, message, command_content):
+        video_metadata = None
+
+        # This looks pretty ugly
+        try:
+            video_metadata = pafy.new(command_content)
+        except ValueError as e:
+            search_result = VideosSearch(command_content).result()
+            video_metadata = pafy.new(search_result["result"][0]["id"])
+
+        audio_url = video_metadata.getbestaudio().url
+
+        if self.voice_client is None:
+            self.voice_client = await message.author.voice.channel.connect()
+
+        if self.voice_client.is_playing():
+            self.voice_client.stop()
+
+        audio_source = discord.FFmpegPCMAudio(audio_url)
+        self.voice_client.play(audio_source)
+        await asyncio.sleep(10)
+
+    async def stop(self, message, command_content):
+        if self.voice_client:
+            self.voice_client.stop()
+
+    async def pause(self, message, command_content):
+        if self.voice_client:
+            self.voice_client.pause()
+
+    async def resume(self, message, command_content):
+        if self.voice_client:
+            self.voice_client.resume()
+
+    async def skip(self, message, command_content):
+        pass
```

Comment:
play/pause/resume/skip need some kind of feedback to the user in the text channel. This can be implemented later, though.

Reply:
Yes at some point